Bezier.cpp

```diff
@@ -6,7 +6,72 @@ Bezier::Bezier(const Vector2& p0, const Vector2& p1, const Vector2& p2, const Ve
 {
 }
 
+//Draw a quadratic bezier curve using segments
 void Bezier::draw(SDL_Renderer *renderer) const
 {
-	// TODO
+	Vector2 curvePoint;
+	Vector2 previousCurvePoint;
+
+	for (float i = 0; i <= segments; i++)
+	{
+		curvePoint = findCurvePoint(i / segments, curvePoint);
+		if (i >= 1)
+		{
+			//Draw the segment line
+			SDL_SetRenderDrawColor(renderer, 100, 100, 100, 255);
+			SDL_RenderDrawLine(renderer, previousCurvePoint.x, previousCurvePoint.y, curvePoint.x, curvePoint.y);
+		}
+		previousCurvePoint = curvePoint;
+	}
+}
+
+//Find the length of the current segment line
+float Bezier::getCurrentSegmentLength(Vector2 currentPosition, SDL_Renderer *renderer) const
+{
+	Vector2 curvePoint;
+	Vector2 previousCurvePoint;
+
+	float segmentLength = 0.0f;
+
+	for (float i = 0; i <= approximateSegments; i++)
+	{
+		curvePoint = findCurvePoint(i / approximateSegments, curvePoint);
+		if (i >= 1)
+		{
+			//If current position is on the segment line
+			if (distance(curvePoint, currentPosition) < distance(previousCurvePoint, currentPosition))
+			{
+				//Get segment length
+				segmentLength = distance(curvePoint, previousCurvePoint);
+				//Draw current segment
+				SDL_SetRenderDrawColor(renderer, 0, 255, 0, 255);
+				SDL_RenderDrawLine(renderer, previousCurvePoint.x, previousCurvePoint.y, curvePoint.x, curvePoint.y);
+			}
+		}
+		previousCurvePoint = curvePoint;
+	}
+
+	return segmentLength;
+}
+
+//Find the distance between two points
+float distance(Vector2 point1, Vector2 point2)
+{
+	return sqrt(pow(point1.x - point2.x, 2) + pow(point1.y - point2.y, 2));
+}
+
+//Applies the quadratic bezier formula to find the current curve point
+Vector2 Bezier::findCurvePoint(float t, Vector2 point) const
```

Comment:
The value of `point` doesn't seem to be used -- it's immediately overwritten, the parameter is passed by value so the changes are local to the function, and the value is returned anyway.